CONTRIBUTING.md

```diff
@@ -16,6 +16,45 @@ As described in the [spec][spec] the build will automatically update the `dist`
 $ npm run build
 ```
 
+## Extract Method
+
+Using this method would only be required after an update to the CLDR database, or if you determine that a language variant file is missing and you would like to create a template for that variant.
+
+To automatically recreate unvalidated language files, perform the following steps:
+
+* Ensure all validated languages have been included in the `src/validated-languages.json` file.
+  * This JSON file must only include the root language name (based on the ISO 639-1 and ISO 639-2 language; meaning no dashes in the folder name).
+  * Any IETF language tag variants that do not match a folder name (e.g. `de-CH` or `fr-CA`) will be ignored.
+
+    ```js
+    [
+      "de",
+      "es",
+      "tr",
+      "ar"
+    ]
+    ```
+
+* For variants:
+  * The extract script will automatically compare the CLDR data between a root language and language variant. If equal, a variant file will not be created. It's not a perfect method so some variants may be missed.
+  * At the time of this writing, no variant files have been determined to be necessary; but `de-CH` (Swiss Standard German) variant was determined to be necessary and has been manually added.
+  * If you determine that a language variant needs to be included, add its IETF language tag to the `src/extract-variants.json` file to force this script to build an "unvalidated" template file for that variant. For example, adding `fr-CA`, the French Canadian language variant to the template file will automatically build a `ca.json` file within the `fr` folder.
+
+    ```js
+    [
+      "fr-CA"
+    ]
+    ```
+
```

Comment:
I don't understand the documentation until here. If I don't understand it, users will probably don't understand it too. I don't understand if the folder name should be a ISO 639-1 name (two characters) or ISO 639-2 (three characters) or even IETF (two characters or including a dash). I also don't understand why you're mentioning language variants for folder names respectively for the validated-languages.json, as it only contains folder names based on _root_ languages, so no variant names (`Any IETF language tag variants that do not match a folder name will be ignored.`).